services/web-app/components/grid-transform/column.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Column as CarbonColumn } from '@carbon/react'
+import clsx from 'clsx'
+import PropTypes from 'prop-types'
+
+import styles from './grid.module.scss'
+
+const Column = ({
+  children,
+  className,
+  colSm,
+  colMd,
+  colLg,
+  colXl,
+  colMax,
+  offsetSm,
+  offsetMd,
+  offsetLg,
+  offsetXl,
+  offsetMax,
+  ...props
+}) => {
+  let sm = ''
+  if (offsetSm !== undefined) {
+    sm = { span: colSm, offset: offsetSm }
+  } else {
+    sm = colSm
+  }
+
+  let md = ''
+  if (offsetLg !== undefined) {
+    md = { span: colMd, offset: offsetMd }
+  } else {
+    md = colMd
+  }
+
+  let lg = ''
+  if (offsetLg !== undefined) {
+    lg = { span: colLg, offset: offsetLg }
+  } else {
+    lg = colLg
+  }
+
+  let xlg = ''
+  if (offsetXl !== undefined) {
+    xlg = { span: colXl, offset: offsetXl }
+  } else {
+    xlg = colXl
+  }
+
+  let max = ''
+  if (offsetMax !== undefined) {
+    max = { span: colMax, offset: offsetMax }
+  } else {
+    max = colMax
+  }
+
+  return (
+    <CarbonColumn
+      sm={sm}
+      md={md}
+      lg={lg}
+      xlg={xlg}
+      max={max}
+      {...props}
+      className={clsx(styles.column, className)}
+    >
+      {children}
+    </CarbonColumn>
+  )
```

Comment:
Refactor suggestion
```suggestion
  const colSizes = {
    sm: offsetSm ? { span: colSm, offset: offsetSm } : colSm,
    md: offsetMd ? { span: colMd, offset: offsetMd } : colMd,
    lg: offsetLg ? { span: colLg, offset: offsetLg } : colLg,
    xlg: offsetXl ? { span: colXl, offset: offsetXl } : colXl,
    max: offsetMax ? { span: colMax, offset: offsetMax } : colMax
  }

  // remove 'noGutter' props to avoid react console error
  const cleanProps = Object.keys(props).filter(key =>
    !key.startsWith('noGutter')).reduce((obj, key) => {
    obj[key] = props[key]
    return obj
  }, {})

  return (
    <CarbonColumn
      {...colSizes}
      {...cleanProps}
      className={clsx(styles.column, className)}
    >
      {children}
    </CarbonColumn>
  )
```